library/src/main/java/com/raizlabs/android/dbflow/sql/SqlUtils.java

```diff
@@ -409,7 +409,10 @@ void delete(final TableClass model, AdapterClass modelAdapter, boolean async) {
      * @param table  The table of the model
      */
     public static void notifyModelChanged(Class<? extends Model> table, BaseModel.Action action) {
-        FlowManager.getContext().getContentResolver().notifyChange(getNotificationUri(table, action), null, true);
+        SQLiteDatabase db = FlowManager.getDatabaseForTable(table).getWritableDatabase();
+        if (!db.inTransaction()) {
```

Comment:
the transaction is commit after we call ```database.endTransaction();```, then the change is commit to the database.

if we query a model in transaction, the data in database is still old, we get the old models, we always want to get the new data when is notified, but the notify in transaction make us query the old model again.

```
        try {
           model.save();
           new Select().from(model.getClass()).query(); // the data is still old
            database.setTransactionSuccessful();
        } finally {
            database.endTransaction();
            SqlUtils.notifyModelChanged(table, null);
        }
```